### ISSUE TITLE
sysrepo application with indirect approach

### ISSUE THREAD

**Issue by biswajit84** [sysrepo/sysrepo#3294]
Hi,
As per my understanding, oper_data_pull_example.c present in sysrepo-2.2.170/examples directory is direct approach.
If my understanding is correct then I need input how to add support for indirect approach. If any documentation/Steps to be added in the same sample code will be really helpful.

**Comment by michalvasko** [comment]
I assume you are talking about the [main page](https://netopeer.liberouter.org/doc/sysrepo/master/html/index.html) of the docs. `oper_data_pull_example.c` does not really represent any of the approaches, it is an example usage of a single function. An example of `indirect approach with sysrepo-plugind` is in the [docs](https://netopeer.liberouter.org/doc/sysrepo/master/html/srpd_example.html). If you do not want to use `sysrepo-plugind`, you can simply add `sr_connect()` and `sr_session_start()` to get the parameters of the plugin init function and you have a standalone daemon.

**Comment by biswajit84** [comment]
Thanks for your input.
As per my understanding, 
Scenario1:if we are trying to connect sysrepo with existing application then indirect approach is best.
Scenario2 : if we are at the very beginning stage of development, then  direct approach is the best.
 We are falling in 2nd scenario. But as you said oper_data_pull_example.c doesn't represent any of the approach.
As sr_connect() and sr_session_start() getting called in the oper_data_pull_example.c so getting confused about direct approach steps to be followed.  If you can share direct approach steps to be followed then it will be helpful.

**Comment by michalvasko** [comment]
I think you are trying too hard to follow some theoretical concept. They are mentioned just so that new users have an idea on how to design their integration with *sysrepo*. In your case I would suggest you simply call whatever *sysrepo* functions you need directly in your app and you should be fine (which is actually the direct approach).

**Comment by biswajit84** [comment]
Understood ..thanks alot for your input.
